selfservice/strategy/oidc/strategy_test.go

```diff
@@ -470,6 +479,67 @@ func TestStrategy(t *testing.T) {
 		return id
 	}
 
+	t.Run("case=force PKCE", func(t *testing.T) {
```

Comment:
Here you can conveniently add tests that check if more complex flows also pass:

1. Sign in with wrong provider, then try again with another
2. Does PKCE actually get validated? Expect an error case
3. Does the automatic detection of whether to sign in or up with PKCE work as well?

Reply:
1. We already have a bunch of these tests in the context of login hints/account linking. I've added a test which checks a mismatch between providers in URL and state is correctly detected and the flow aborted.
2. Added tests.
3. Test is already there.